topics/request.md

```diff
@@ -1,316 +1,116 @@
-with the most common HTTP verbs: `GET`, `POST`, `PUT`, `DELETE`, `PATCH`, `HEAD` and `OPTIONS`.
-
+### Headers {id="headers"}
+To add headers to the request, use the [headers](https://api.ktor.io/%ktor_version%/io.ktor.client.request/-http-request-builder/headers.html) function as follows:
 ```kotlin
-val text = client.post<String>("http://127.0.0.1:8080/")
 ```
+{src="snippets/_misc_client/GetMethodWithHeaders.kt"}
 
-When calling request methods, you can provide a lambda to build the request
-parameters like the URL, the HTTP method(verb), the body, or the headers:
 
-```kotlin
-val text = client.post<String>("http://127.0.0.1:8080/") {
-    header("Hello", "World")
-}
-```
 
-The [HttpRequestBuilder](https://api.ktor.io/%ktor_version%/io.ktor.client.request/-http-request-builder/) looks like this:
+### Cookies {id="cookies"}
+To send cookies, use the [cookie](https://api.ktor.io/%ktor_version%/io.ktor.client.request/cookie.html) function:
 
 ```kotlin
-class HttpRequestBuilder : HttpMessageBuilder {
-    var method: HttpMethod
-
-    val url: URLBuilder
-    fun url(block: URLBuilder.(URLBuilder) -> Unit)
-
-    val headers: HeadersBuilder
-    fun header(key: String, value: String)
-    fun headers(block: HeadersBuilder.() -> Unit)
-
-    var body: Any = EmptyContent
-
-    val executionContext: CompletableDeferred<Unit>
-    fun setAttributes(block: Attributes.() -> Unit)
-    fun takeFrom(builder: HttpRequestBuilder): HttpRequestBuilder
-}
 ```
+{src="snippets/_misc_client/GetMethodWithCookies.kt"}
 
-The `HttpClient` class only offers some basic functionality, and all the methods for building requests are exposed as extensions.\\
-You can check the standard available [HttpClient build extension methods](https://api.ktor.io/%ktor_version%/io.ktor.client.request/).
-
-### Customize method
-
-In addition to call, there is a `request` method for performing a typed request,
-[receiving a specific type](response.md#receive) like String, HttpResponse, or an arbitrary class.
-You have to specify the URL and the method when building the request.
-
-```kotlin
-val call = client.request<String> {
-    url("http://127.0.0.1:8080/")
-    method = HttpMethod.Get
-}
-```
+Note that Ktor provides the [](http-cookies.md) feature that allows you to keep cookies between calls.
 
-### Posting forms
 
-{id="submit-form"}
 
-There are a couple of convenience extension methods for submitting form information.
-The detailed reference is listed [here](https://api.ktor.io/%ktor_version%/io.ktor.client.request.forms/).
 
-The `submitForm` method:
+## Set request body {id="body"}
+To set the body of a request, you need to specify the [body](https://api.ktor.io/%ktor_version%/io.ktor.client.request/-http-request-builder/body.html) property exposed by `HttpRequestBuilder`. This property accepts different types of payload, including plain text, arbitrary class instances, form data, byte arrays, and so on. Below we'll take a look at several examples.
```

Comment:
I'd not use verb exposed here. I'd just say property via ..

payload > payloads